set minimal HIP version to 3.5

psychocoderHPC opened the pull request:
- Update to the FindHIP implementation from rocm-3.5
- set HIP 3.5 as minimum requirement
- switch CI base container for HIP to rocm3.5 container
  - test HIP-clang instead of HIP-nvcc
  - remove testing for HIP-nvcc

psychocoderHPC committed:
update CMake FindHIP

Update to the FindHIP implementation from rocm-3.5

psychocoderHPC commented on .github/workflows/ci.yml:
```diff
@@ -74,10 +74,10 @@ jobs:
         ## HIP
         - name: linux_hip_nvcc-9.2_gcc-5_debug_hip_only
-          os: ubuntu-latest
+          os: rocm/pytorch:rocm3.5_ubuntu16.04_py3.6_pytorch
```
@BenjaminW3 I expect this is the place to load a different base container or? I will switch to a container with clean installed hip-clang.

BenjaminW3 replied:
I do not think that this works. The `os:` only allows one of the few virtual machine types which github allows.

BenjaminW3 replied:
`ALPAKA_CI_DOCKER_BASE_IMAGE_NAME` is the correct one

psychocoderHPC replied:
thx

psychocoderHPC committed:
switch to HIP rocm docker container

- use HIP-clang instead of HIP-nvcc
- remove hip install script
- remove not required environment variables

psychocoderHPC committed:
remove purge section

The if section is not required because we do not build container.

psychocoderHPC commented:
@BenjaminW3 I switched now to a rocm container and we test HIP-clang. HIP-clang was always the target for HIP and since rocm3.5 is shipping HIP-clang we do not handle installation.

BenjaminW3 commented on script/install.sh:
```diff
@@ -67,7 +66,7 @@ fi
+    sudo apt-get --quiet --purge --yes autoremove
     sudo apt-get clean
     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 fi
```
You can delete this whole `if-fi` block because it is not useful as we do not cache the image anymore.

psychocoderHPC replied:
will do

psychocoderHPC commented:
I removed the if block and CI passed, ready to merge
